shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-dialect/shardingsphere-data-pipeline-mysql/src/test/java/org/apache/shardingsphere/data/pipeline/mysql/check/consistency/CRC32MatchMySQLSingleTableDataCalculatorTest.java

```diff
@@ -0,0 +1,85 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.data.pipeline.mysql.check.consistency;
+
+import org.apache.shardingsphere.data.pipeline.api.check.consistency.DataCalculateParameter;
+import org.apache.shardingsphere.data.pipeline.api.datasource.PipelineDataSourceWrapper;
+import org.apache.shardingsphere.data.pipeline.core.exception.PipelineDataConsistencyCheckFailedException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.StreamSupport;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.RETURNS_DEEP_STUBS;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class CRC32MatchMySQLSingleTableDataCalculatorTest {
+
+    @Mock
+    private DataCalculateParameter dataCalculateParameter;
+
+    private PipelineDataSourceWrapper pipelineJobPrepareFailedException;
+
+    @Before
+    public void setUp() throws SQLException {
+        pipelineJobPrepareFailedException = mock(PipelineDataSourceWrapper.class, RETURNS_DEEP_STUBS);
+        Collection<String> columnNames = Arrays.asList("fieldOne", "fieldTwo", "fieldThree");
+        when(dataCalculateParameter.getLogicTableName()).thenReturn("tableName");
+        when(dataCalculateParameter.getColumnNames()).thenReturn(columnNames);
+        when(dataCalculateParameter.getDataSource()).thenReturn(pipelineJobPrepareFailedException);
+    }
+
+    @Test
+    public void assertCRC32MatchMySQLSingleTableDataCalculatorSuccess() {
+        String algorithmType = new CRC32MatchMySQLSingleTableDataCalculator().getAlgorithmType();
+        assertThat(algorithmType, is("CRC32_MATCH"));
+    }
+
+    @Test
+    public void assertGetDatabaseTypesSuccess() {
+        Collection<String> databaseTypes = new CRC32MatchMySQLSingleTableDataCalculator().getDatabaseTypes();
+        assertThat(databaseTypes.size(), is(1));
+        assertThat(databaseTypes.stream().findFirst().get(), is("MySQL"));
+    }
+
+    @Test
+    public void assertCalculateSuccess() {
+        Iterable<Object> calculate = new CRC32MatchMySQLSingleTableDataCalculator().calculate(dataCalculateParameter);
+        long calculateSize = StreamSupport.stream(calculate.spliterator(), false).count();
+        assertThat(calculateSize, is(3L));
```

Comment:
Could `3L` be replaced to `columnNames.size()`? So if `columnNames` value changed, it's not necessary to change the hard coded `3L`.